src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -422,6 +425,113 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
         }
     }
 
+    public static boolean correctPolyAmbiguity(GeoPoint[] points, boolean handedness) {
+        return correctPolyAmbiguity(points, handedness, computePolyOrientation(points), 0, points.length, false);
+    }
+
+    public static boolean correctPolyAmbiguity(GeoPoint[] points, boolean handedness, boolean orientation, int component, int length,
+                                               boolean shellCorrected) {
+        // OGC requires shell as ccw (Right-Handedness) and holes as cw (Left-Handedness)
+        // since GeoJSON doesn't specify (and doesn't need to) GEO core will assume OGC standards
+        // thus if orientation is computed as cw, the logic will translate points across dateline
+        // and convert to a right handed system
+
+        // compute the bounding box and calculate range
+        Pair<Pair, Pair> range = GeoUtils.computeBBox(points, length);
+        final double rng = (Double)range.getLeft().getRight() - (Double)range.getLeft().getLeft();
+        // translate the points if the following is true
+        //   1.  shell orientation is cw and range is greater than a hemisphere (180 degrees) but not spanning 2 hemispheres
+        //       (translation would result in a collapsed poly)
+        //   2.  the shell of the candidate hole has been translated (to preserve the coordinate system)
+        boolean incorrectOrientation = component == 0 && handedness != orientation;
+        boolean translated = ((incorrectOrientation && (rng > DATELINE && rng != 360.0)) || (shellCorrected && component != 0));
+        if (translated) {
+            for (GeoPoint c : points) {
+                if (c.x < 0.0) {
+                    c.x += 360.0;
+                }
+            }
+        }
+        return translated;
+    }
+
+    public static boolean computePolyOrientation(GeoPoint[] points) {
+        return computePolyOrientation(points, points.length);
+    }
+
+    public static boolean computePolyOrientation(GeoPoint[] points, int length) {
+        // calculate the direction of the points:
+        // find the point at the top of the set and check its
+        // neighbors orientation. So direction is equivalent
+        // to clockwise/counterclockwise
+        final int top = computePolyTop(points, length);
+        final int prev = ((top + length - 1) % length);
+        final int next = ((top + 1) % length);
+        return (points[prev].x > points[next].x);
+    }
+
+    private static final int computePolyTop(GeoPoint[] points, int length) {
```

Comment:
i still feel like we should make this function name "correct" and maybe call it "computePolyBottomLeft"?